scenario_simulation/scripts.py

```diff
@@ -21,26 +24,29 @@ def estimate_reward(
 
     # Do n scenario simulations
     for i in range(number_of_simulations):
-        iteration_counter = 0
-        child_state = copy.deepcopy(state)
-        total_reward = 0
-
+        simulation_counter = 1
+        world = classes.World(
+            remaining_shift_duration, initial_state=copy.deepcopy(state)
+        )
+        next_is_vehicle_action = True
         # Simulate until shift ends
-        while iteration_counter * ITERATION_LENGTH_MINUTES < remaining_shift_duration:
-            iteration_counter += 1
-            # all possible actions in this state
-            possible_actions = child_state.get_possible_actions(number_of_neighbours=3)
-
-            # pick a random action
-            random_action = possible_actions[
-                np.random.randint(0, len(possible_actions))
-            ]
-            total_reward += child_state.do_action(random_action)
-
-            _, _, lost_demand = child_state.system_simulate()
-
-            total_reward -= lost_demand * LOST_TRIP_REWARD
-
-        all_rewards.append(total_reward)
+        while world.time < remaining_shift_duration:
+            if next_is_vehicle_action:
+                action = policies.RandomActionPolicy.get_best_action(world)
+                world.add_reward(world.get_discount() * world.state.do_action(action))
+                world.time = world.time + action.get_action_time(
+                    world.state.get_distance_id(
+                        world.state.current_cluster.id, action.next_cluster
+                    )
+                )
+            else:
+                _, _, lost_demand = world.state.system_simulate()
+                world.add_reward(-lost_demand * LOST_TRIP_REWARD)
```

Comment:
LOST_TRIP_REWARD is negative. So here we get double negative -> and that gives positive